### DISCUSSION TITLE
Memory usage difference between trained and pretrained VITS model

### DISCUSSION THREAD

**Discussion by MahdiEsrafili** [coqui-ai/TTS#1890] [General Q&A]
Hi.  When I test officially released VITS model on cpu on long sentences, it used about 4-5 GB of RAM . But When I test my own trained VITS model, it uses more than 16 GB of memory. So how can I make memory usage of my model same as official models?
This is my config file:
[config.zip](https://github.com/coqui-ai/TTS/files/9386504/config.zip)

**Comment by erogol** [comment]
If you remove the discriminator model, you would get much lesser RAM use.

**Comment by MahdiEsrafili** [comment]
I use this code to remove discriminator model and optimizer:
```python
import torch
from TTS.tts.models.vits import Vits
from TTS.config import load_config
from TTS.tts.utils.text.tokenizer import TTSTokenizer
from TTS.utils.audio import AudioProcessor

tts_checkpoint="../../model/checkpoint_3870000.pth"
tts_config_path="../../model/config.json"
save_checkpoint = "../../model/model.pth"

config = load_config(tts_config_path)
ap = AudioProcessor.init_from_config(config)
tokenizer, config = TTSTokenizer.init_from_config(config)

# init model
model = Vits(config, ap, tokenizer, speaker_manager=None)
model.load_checkpoint(config, tts_checkpoint, eval=True)
model.disc = None
model_state = model.state_dict()
state = {
    "model": model_state
    }
torch.save(state, save_checkpoint)
```
This decreases model size from 900MB to140MB. I also set `init_discriminator ` to `false` in config file. But when I synthesize a text including 90 words on cpu, it uses more than 12 GB of RAM. Also about 2GB of RAM synthesizing a text with 10 words (5.7 seconds audio). Is it OK or I'm doing something wrong?

**Reply by iamkhalidbashir** [reply]
Interested to know if you found the solution for this

**Reply by MahdiEsrafili** [reply]
@iamkhalidbashir Not a complete fix yet

**Reply by Raphi1** [reply]
I'm also facing the issue with high memory usage and big model size. My model is almost 1GB. I'm using the vits recipe from the GitHub repository. For training, I used the Google colab. If I'm initializing the model like above, it fails to train, as NoneObject is not iterable. The `init_discriminator` is being ignored from the config. It's always true in the new run directory. I also tried to add the VitsArgs to put that value in the VitsConfig. Then init_discriminator will be set correctly, but again the training fails. I train the model based on Thorsten voice tts model.
As I want to run the tts-server on a Raspberry Pi and it's already running with the pre trained models, I'm looking forward to get some help achieving this goal :). Is it possible, to remove the unused data after training, as I already have a trained model?

Thanks
